index.html

```diff
@@ -439,10 +439,9 @@ <h2>
       </h2>
       <p data-link-for="WebAppManifest">
         A <dfn data-export="">navigation scope</dfn> is a <a>URL</a> that
-        represents the set of URLs to which an <a>application context</a> can
-        be navigated while the manifest is <a>applied</a>. The <a>navigation
-        scope</a> of a manifest <var>manifest</var> is
-        <var>manifest</var>["<a>scope</a>"].
+        represents the set of URLs which are considered to be part of an
```

Comment:
> URLs outside the scope can still be part of the web application (e.g., an about.html page that opens in an overlay window), it's just that the manifest members are no longer in effect there.

But the manifest members _are_ in effect there. That's exactly why I'm making this change. When you're off-scope, all of the manifest members are still affecting the application context, because we want it to still feel like you're in the app.

Are you disagreeing with the _behaviour_ allowed by this change, or just the wording? i.e., do you think Chrome should not render about.html page in an overlay window with the full theming of the app manifest behind it (the theme colour, app name, icon, display mode, etc)? Or do you just disagree on the semantics?

Reply:
> Are you disagreeing with the behaviour allowed by this change, or just the wording?

Just the wording. Agree about the behavior. 

With the wording, I think it risks overstepping into defining "what is/is not a web application"... the original definition was just about the "manifest applies to set of URLs"

Reply:
OK maybe we can workshop this. I don't want to return to the old definition, since it directly ties the scope to the set of URLs that the manifest "applies" to, where "applies" has a strict technical definition of having all of its members used by the top-level browsing context (and that can now be true for all URLs).

It wasn't my intention to try and define a web application here. What's really important here is three things:

1. A "navigation scope" is just a URL.
2. Really though, the URL represents a set of URLs, which is all the URLs that return true from the "within scope" algorithm.
3. The actual normative meaning of the navigation scope is defined by every other thing that uses the "navigation scope" or "within scope" concept.

So 2 and 3 are taken care of elsewhere. All that this definitional sentence really needs to do is say "it's a URL". I thought it was a bit empty, so I added "it represents the set of URLs which are considered to be part of an application". That feels a bit non-normative, so we can take it out, and just give the facts about "navigation scope" as if we were defining a typedef in a computer program, and not give its meaning in this statement. Simply: "A navigation scope is a URL."

Reply:
> OK maybe we can workshop this. I don't want to return to the old definition, since it directly ties the scope to the set of URLs that the manifest "applies" to, where "applies" has a strict technical definition of having all of its members used by the top-level browsing context (and that can now be true for all URLs).

I'm confused why "that can now be true for all URLs"? 

> So 2 and 3 are taken care of elsewhere. All that this definitional sentence really needs to do is say "it's a URL".

> "A navigation scope is a URL.

Although the above is true (it's a URL indeed), that doesn't really define the role of the navigation scope as it relates to applying the manifest. That needs to be either here, or somewhere that says roughly "when you navigate, keep applying the manifest until you are no longer within the navigation scope".

Reply:
(FYI I'm leaving this on hold until #882 lands, which will change the definition of "navigation scope", though it keeps the clause in question.)

> I'm confused why "that can now be true for all URLs"?

The current definition defines navigation scope as "the set of URLs to which an application context can be navigated while the manifest is applied." As of #701 (which landed nearly two years ago), the set of URLs to which an application context can be navigated while the manifest is applied is all URLs. If I navigate an app window on https://mysite.com to https://yoursite.com, the manifest is still _applied_.

So we can't have this definition, if the intention is to allow the manifest to be applied when you navigate out of scope. (This isn't directly contradicted by normative text, but the absence of any text that forbids out-of-scope navigation implies that you can navigate out of scope in an application context where the manifest is applied. A non-normative note directly contradicts this definition: "Nothing prevents an application context from navigating to a URL that is outside of the application's navigation scope , while still having the manifest applied to it.")

> Although the above is true (it's a URL indeed), that doesn't really define the role of the navigation scope as it relates to applying the manifest. That needs to be either here, or somewhere that says roughly "when you navigate, keep applying the manifest until you are no longer within the navigation scope".

Wait: "until you are no longer within the navigation scope" implies that you would not apply the manifest when navigated off-scope. Do we have a disagreement over the meaning of the term "apply"? I am considering a manifest to be applied permanently to the application context, regardless of whether it's navigated off-scope. When you navigate off-scope, it is RECOMMENDED that the application window shows some UI indicating that you are off scope, but the manifest is still applied. We would still show the theme colour, title and icon of the application from the manifest. So the spec needs to state that the manifest is always applied to both in-scope and off-scope documents.

I think we can add some text to "applying the manifest" that says "When a manifest is applied to a top-level browsing context, it is continually applied throughout the lifetime of the browsing context."

Reply:
Discussed on call... manifest members MAY apply outside the navigation scope (e.g., theme color). We also discussed that a new/independent top-level browsing context is not created.

Reply:
We just discussed this in our F2F. Agreed that the manifest _is_ applied to the out-of-scope page, based on testing Chrome, Firefox and Safari (the last of those we're not 100% sure but we suspect that's happening).

There isn't any normative text that says this, but it's implied by the fact that we do not make changes to the HTML navigation system, therefore navigation is expected to proceed as normal to an out-of-scope URL.

There is non-normative text that supports this:

> Unlike previous versions of this specification, user agents are no longer required or allowed to block off-scope navigations, or open them in a new top-level browsing context. This practice broke some sites that navigate to an off-scope URL (e.g., to perform third-party authentication). See Issue 646.

I proposed that since it's been 2 years, we collapse this text to not mention the history of it, just say that user agents are not allowed to block or transfer the navigation to another TLBC.

@marcoscaceres let me know if I misunderstood something.

Reply:
That sounds right to me. Thanks for the summary.